src/test/java/org/broadinstitute/hellbender/tools/dragstr/InterleavingListSpliteratorUnitTest.java

```diff
@@ -0,0 +1,286 @@
+package org.broadinstitute.hellbender.tools.dragstr;
+
+import org.apache.commons.math3.util.Pair;
+import org.broadinstitute.hellbender.utils.Utils;   
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+public class InterleavingListSpliteratorUnitTest {
+
+    enum Operation {
+        // trySplit
+        SPLIT('^'),
+        // tryAdvance
+        ADVANCE('.'),
+        // forEachRemaining.
+        READTHRU('*'),
+        // popEmpty.
+        POPEMPTY('$');
+
+        final char code;
+
+        Operation(final char code) {
+            this.code  = code;
+        }
+
+        static Operation of(char ch) {
+            if (ch == SPLIT.code) {
+                return SPLIT;
+            } else if (ch == ADVANCE.code) {
+                return ADVANCE;
+            } else if (ch == READTHRU.code) {
+                return READTHRU;
+            } else if (ch == POPEMPTY.code) {
+                return POPEMPTY;
+            } else if (Character.isDigit(ch)) {
+                return ADVANCE;
+            } else {
+                throw new IllegalArgumentException();
+            }
+        }
+    }
+
+    @Test(dataProvider = "lengthAndNumberOfThreadsData")
+    public void testParallel(final int length, final int numberOfThreads) {
+        final List<Integer> values = firstNaturalIntegers(length);
+        final List<Integer> result = Utils.runInParallel(numberOfThreads,
+                () -> StreamSupport.stream(new InterleavingListSpliterator<>(values), true)
+                    .collect(Collectors.toList()));
+
+        Collections.sort(result);
+        Assert.assertEquals(result, values);
+    }
+
+    @Test(dataProvider = "lengthAndSplitLevelsData")
+    public void testInterleaveness(final int length, final int splitLevels) {
+        final Spliterator<Integer> subject = new InterleavingListSpliterator<>(firstNaturalIntegers(length));
```

Comment:
This test actually looks good. Looking at the expected values in the debugger it looks correct to me and its a nice test of all the various options this spliterator can hit.

Reply:
thanks.